src/main/java/com/liferay/faces/util/i18n/internal/I18nImpl.java

```diff
@@ -53,10 +55,27 @@ public I18nImpl() {
 
 		// Store the resource bundle cache in the application map (as a Servlet Context attribute).
 		if (startupFacesContext != null) {
+
 			ExternalContext externalContext = startupFacesContext.getExternalContext();
 			Map<String, Object> applicationMap = externalContext.getApplicationMap();
-			Map<Locale, ResourceBundle> cache = new ConcurrentHashMap<Locale, ResourceBundle>();
-			applicationMap.put(I18nImpl.class.getName(), cache);
+			Cache<Locale, ResourceBundle> facesResourceBundleCache;
+
+			String maxCacheCapacityString = externalContext.getInitParameter(I18n.class.getName() +
+					".maxCacheCapacity");
+
+			if (maxCacheCapacityString != null) {
+
+				CacheFactory cacheFactory = (CacheFactory) FactoryExtensionFinder.getFactory(externalContext,
+						CacheFactory.class);
+				int initialCacheCapacity = cacheFactory.getDefaultInitialCapacity();
```

Comment:
@ngriffin7a, that seems like a nice solution, so we can get rid of that method and make the default still customizable. Loses some flexibility and it's a less obvious, but since we have so many examples and it means less java API, I'm all for adding a `WebConfigParam` for `DefaultInitialCacheCapacity` (or `DefaultInitCacheCapacity`).